src/FileCache.php

```diff
@@ -365,9 +362,8 @@ private function getCacheFile(string $key): string
         $base = $this->cachePath;
 
         for ($i = 0; $i < $this->directoryLevel; ++$i) {
-            if (($prefix = substr($key, $i + $i, 2)) !== false) {
-                $base .= DIRECTORY_SEPARATOR . $prefix;
-            }
```

Comment:
From PHP 8 we should check on empty string instead false. Seems, need test for this case (in this PR or create issue).